iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java

```diff
@@ -143,12 +145,19 @@ public TSStatus setDatabase(DatabaseSchemaPlan databaseSchemaPlan) {
 
     try {
       clusterSchemaInfo.isDatabaseNameValid(databaseSchemaPlan.getSchema().getName());
+      if (!databaseSchemaPlan.getSchema().getName().equals(IoTDBConfig.SYSTEM_DATABASE)) {
+        clusterSchemaInfo.checkDatabaseLimit();
+      }
```

Comment:
I think we need to put this check into StateMachine to guarantee the atomicity of `check` and `create`.